spec/requests/requests_spec.rb

```diff
@@ -269,8 +269,7 @@
         "id"               => request.id.to_s,
         "workflow"         => a_hash_including("values"),
         "v_allowed_tags"   => [a_hash_including("children")],
-        "v_workflow_class" => a_hash_including(
-          "instance_logger" => a_hash_including("klass" => request.workflow.class.to_s))
```

Comment:
I have no idea why this passed on rails 6.1 🤣

Reply:
FYI, instance_logger is an instance variable coming from the Vmdb::Logging stuff.  Not sure why this was coming back previously.  Maybe json dump of the class object included the ivars.  That's why I'm changing it to return the Class#name as it's weird to store the Class object.

Reply:
where is v_workflow_class defined?

Reply:
https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_request.rb#L51

Reply:
The rescue nil in the [class method](https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_request.rb#L339-L347) drives me mad and I think it's related to the NilClass `{}` above.  I really have no idea how any of this stuff works but this `instance_logger` in the test was completely wrong and thankfully rails 7 detected it.  Not sure why it worked in 6.1

Reply:
interestingly, I could only find `v_workflow_class` referenced in this spec. Not sure where else it is used. It is declared as a `String` and should not be a hash.

Reply:
Wow, I had so much trouble tracking down where this method was defined `delegate` is not very `grep` friendly. It was added in https://github.com/ManageIQ/manageiq/pull/13441

```ruby
  virtual_column  :v_workflow_class,     :type => :string,   :uses => :workflow
  delegate :class,                       :to => :workflow,   :prefix => :v_workflow
```

Interestingly, there is a `workflow_class` method that we probably should be using instead.
But more importantly, it is declared to return a `String` (which it probably should be), but it returns a `Class`. It is possible that our serialization of a `Class` may have changed.

It appears that the example is not properly creating a workflow, so `nil.class` is returning a `NilClass`.

---

I put a `byebug` into the class but getting all sorts of errors:

These all work in `rails c` for me, but fail when I'm running this specific spec.

```ruby
(byebug) MiqRequest.new.description
*** NameError Exception: uninitialized constant MiqRequest::TASK_DESCRIPTION

  default_value_for(:message)       { |r| "#{r.class::TASK_DESCRIPTION} - Request Created" }
(byebug) request.class
MiqProvisionRequest
(byebug) MiqProvisionRequest.new.description
nil
(byebug) request.v_workflow_class
*** NameError Exception: uninitialized constant MiqRequestWorkflow::DialogFieldValidation

  include DialogFieldValidation
          ^^^^^^^^^^^^^^^^^^^^^

(byebug) require "miq_request_workflow/dialog_field_validation"
(byebug) MiqProvisionRequest.new.v_workflow_class
(byebug) MiqProvisionRequest.new.workflow_class
^^^ all give the same error
```

I tried redefining the `DialogFieldValidation` to the long style (`module MiqProvisionWorkflow; module DialogFieldValidator; end; end` instead of `module parent::DialogFieldValidator`) but that did not help.

This is a tangent, but this feels a little chicken and egg.
If you want to define the child module, define the parent class (which references the child module)

Reply:
update:

The MiqProvisionRequest.workflow_class references a class, but `constantize` throws that exception listed above. Match that with a `rescue nil` and we get a `nil` coming out. `nil.class == NilClass`

Reply:
yeah, is the test and code busted or just the test?  I can try to create a proper workflow to fix the test.

Reply:
Thnx Jason: once I updated all my repos, I was able to load this just fine.

One thing about the code, it compares the `workflow.class` and the `workflow_class`. So if it gets busted (goes from a value to nil), then it will always pass the test.

Reply:
I'm punting on this (see below).  We'll tackle this if we determine the correct behavior it should have.  Right now, it doesn't make sense why it's ok on rails 6.1 when it's clearly wrong.  It's possible no one is using it and not reported a bug.  Either way, we can handle it when it's a problem.